package/MDAnalysis/analysis/base.py

```diff
@@ -40,17 +40,59 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
```

Comment:
I'd rather this was `_Results` so we're not committed to supporting use of this externally.

Reply:
I will become a hard requirement in the Analysis API if we go forward with this PR — I don't see a way around making this explicit and well defined. And once we do this, we might as well make it public.

In order to be compatible with mdacli, all analysis classes _must_ put results into this data structure (right @PicoCentauri @joaomcteixeira ?). In order to ensure that all our classes are properly interoperable we will have to require that `Results` (or rather the instance `AnalysisBases.results`) be used by everyone — including future MDKits.

Reply:
My first thought was also to make it private but Oliver is right. The CLI will scan for the existence and only take modules into account with `AnalysisBases.results`. Without a well defined documentation it is probably hard for developers to implement results in their classes.

Reply:
See comment below: define the API in the docs.

Reply:
Is it best to subclass dict or UserDict?

Reply:
https://treyhunner.com/2019/04/why-you-shouldnt-inherit-from-list-and-dict-in-python/#What%E2%80%99s_the_alternative_to_inheriting_from_list_and_dict?

Reply:
Thanks for figuring out. I will adjust the parent class.

Reply:
Thanks @lilyminium , I learned something new.